aea/configurations/base.py

```diff
@@ -1137,23 +1130,19 @@ def _create_or_update_from_json(
 
     def get_overridable(self) -> dict:
         """Get overridable configuration data."""
-        result = {}
-        current_config_data = self.json
-        if self.abstract_field_name in current_config_data:
-            result[self.abstract_field_name] = current_config_data[
-                self.abstract_field_name
-            ]
+        result = super().get_overridable()
 
-        for field in self.FIELDS_WITH_NESTED_FIELDS:
-            if not current_config_data.get(field, {}):
+        for overridable in ("handlers", "behaviours", "models"):
+            if not result.get(overridable):
                 continue
-            result[field] = {}
-            for name in current_config_data[field].keys():
-                result[field][name] = {}
-                for nested_field in self.NESTED_FIELDS_ALLOWED_TO_UPDATE:
-                    result[field][name][nested_field] = current_config_data[field][
-                        name
-                    ][nested_field]
+            for field in result[overridable]:
+                # We don't want users to override the `class_name`` parameter
+                # for a behaviour/handler/model object.
+                del result[overridable][field]["class_name"]
+
```

Comment:
We also need to ensure that `args` still exists.

I am not sure this new approach is more transparent than the old one...

Reply:
The `arg` is going to be there since `result` is just copy of the `SkillConfig.json` and we run this loop to make sure we don't include `class_name` in the overridables https://github.com/valory-xyz/open-aea/blob/31d8c486cbad9a1791196331434cc19674fc7754/aea/configurations/base.py#L1131

Continuing from https://github.com/valory-xyz/open-aea/pull/379#discussion_r998951199  it's not about transparency, it's rather simplifying the previous method which I already explained was an overly complicated way to filter out `class_method` parameter